syntax="proto3";


message RequestUnitMessage {
    string full_name = 1;
    repeated string phone_numbers = 2;
    repeated string emails = 3;
    repeated string addresses = 4;
    string passport_number = 5;
    string birthday = 6;
}

message ValidString {
    bool valid = 1;
    string requested_value = 2;
}

message ValidUnitMessage {
    ValidString full_name = 1;
    repeated ValidString phone_numbers = 2;
    repeated ValidString emails = 3;
    repeated ValidString addresses = 4;
    ValidString passport_number = 5;
    ValidString birthday = 6;
}

message ResponseUnitMessage {
    uint32 guid = 1;
    ValidUnitMessage msg = 2;
}

service ValidationDispenser {
    rpc Validate(RequestUnitMessage) returns (ResponseUnitMessage) {}
}

